src/main/java/io/rcktapp/api/handler/firehose/FirehoseDb.java

```diff
@@ -56,7 +59,24 @@ public FirehoseDb() {
     public void bootstrapApi() throws Exception {
         List<Pair<String, String>> nameActuals = new ArrayList<>();
         // our local streams
-        getFirehoseClient().listDeliveryStreams(new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut")).getDeliveryStreamNames().forEach(name -> nameActuals.add(Pair.of(name.toLowerCase(), name)));
+        // getFirehoseClient().listDeliveryStreams(new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut")).getDeliveryStreamNames().forEach(name -> nameActuals.add(Pair.of(name.toLowerCase(), name)));
+
+        ListDeliveryStreamsRequest listDeliveryStreamsRequest = new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut");
+        listDeliveryStreamsRequest.setLimit(listDeliveryStreamsLimit);
+        ListDeliveryStreamsResult listDeliveryStreamsResult = getFirehoseClient().listDeliveryStreams(listDeliveryStreamsRequest);
+        List<String> deliveryStreamNames = listDeliveryStreamsResult.getDeliveryStreamNames();
+
+        while (listDeliveryStreamsResult.getHasMoreDeliveryStreams())
```

Comment:
Wouldn't it make more sense for this to be a do-while instead of a while, as you have to run a request BEFORE you know whether you need to do it again?